drivers/dma/bcm2835-dma.c

```diff
@@ -284,7 +290,7 @@ struct bcm2835_desc {
 /* the max dma length for different channels */
 #define MAX_DMA40_LEN SZ_1G
 
-#define BCM2711_DMA40_BURST_LEN(x)	((min(x,16) - 1) << 8)
+#define BCM2711_DMA40_BURST_LEN(x)	(((x) & 15) << 8)
```

Comment:
This has changed behaviour. Was it just totally wrong to start with, or something more subtle?

Reply:
Just to clarify - the dma40 code has never been run before (in locks up kernel in any possibly use), so regressions aren't an issue.

This is semantics.  A burst describes the number of additional beats added to the original access.
Storing 0 in this field would just be the original access.
Storing 3 in this field would be original access plus 3 extra beats.

I think it makes sense to consider this fields as "extra burst beats", rather than "total bust beats".
That is how the non-dma40 code works, and how the hardware works.